Source/MQTTnet/PayloadSegmentExtensions.cs

```diff
@@ -0,0 +1,64 @@
+using MQTTnet.Internal;
+using System;
+using System.Runtime.InteropServices;
+
+namespace MQTTnet
+{
+    public static class PayloadSegmentExtensions
+    {
+        private static readonly ArraySegment<byte> emptySegment = new ArraySegment<byte>(EmptyBuffer.Array);
+
+        /// <summary>
+        /// Get the ArraySegment style of Payload
+        /// </summary>
+        /// <param name="segmentable"></param>
+        /// <returns></returns>
+        public static ArraySegment<byte> GetPayloadSegment(this IPayloadSegmentable segmentable)
+        {
+            if (segmentable.Payload == null)
+            {
+                return emptySegment;
+            }
+
+            var payloadCount = segmentable.PayloadCount == null
```

Comment:
```suggestion
            var payloadLength = segmentable.PayloadLength == null
```
and all other places as well.